kore/src/Kore/Step/SMT/Representation/All.hs

```diff
@@ -46,9 +53,40 @@ sorts).
 build
     :: VerifiedModule Attribute.Symbol
     -> Map.Map Id Attribute.Constructors
-    -> AST.SmtDeclarations
+    -> Step.AST.SmtDeclarations
 build indexedModule sortConstructors =
-    resolve (sorts `AST.mergePreferFirst` symbols)
+    removeDuplicates $ resolve (sorts `Step.AST.mergePreferFirst` symbols)
   where
     sorts = Sorts.buildRepresentations indexedModule sortConstructors
     symbols = Symbols.buildRepresentations indexedModule
+
+removeDuplicates :: Step.AST.SmtDeclarations -> Step.AST.SmtDeclarations
```

Comment:
Could we call this `removeDuplicateConstructorDeclarations` instead? Because it's only removing constructor symbol declarations, and someone might look at it and think there could be other forms of duplication.